File: app/models/email_log.py
from datetime import datetime

from sqlalchemy import DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.datetime_utils import now_sp
from app.database import Base


class EmailLog(Base):
    __tablename__ = "email_logs"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    recipient: Mapped[str] = mapped_column(String, index=True, nullable=False)
    subject: Mapped[str] = mapped_column(String, nullable=False)
    body: Mapped[str] = mapped_column(Text, nullable=False)
    sent_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=now_sp)
    status: Mapped[str] = mapped_column(String, nullable=False) # 'SUCCESS' or 'FAILED'
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)

File: backend/internal/models/email_log.go
package models

import "time"

type EmailLog struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	Recipient    string    `gorm:"index;not null" json:"recipient"`
	Subject      string    `gorm:"not null" json:"subject"`
	Body         string    `gorm:"type:text;not null" json:"body"`
	SentAt       time.Time `gorm:"autoCreateTime" json:"sent_at"`
	Status       string    `gorm:"not null" json:"status"` // 'SUCCESS' or 'FAILED'
	ErrorMessage *string   `gorm:"type:text" json:"error_message"`
}

func (EmailLog) TableName() string { return "email_logs" }
